src/RA_httpthread.cpp

```diff
@@ -252,31 +247,51 @@ void RAWeb::SetUserAgentString()
 #ifndef NTSTATUS
     #define NTSTATUS long
 #endif
-    using fnRtlGetVersion = NTSTATUS(NTAPI*)(PRTL_OSVERSIONINFOEXW lpVersionInformation);
-    auto RtlGetVersion = (fnRtlGetVersion)GetProcAddress(GetModuleHandleW(L"ntdll.dll"), "RtlGetVersion");
-    if (RtlGetVersion)
     {
-        RTL_OSVERSIONINFOEXW osVersion = {};
+        using fnRtlGetVersion = NTSTATUS(NTAPI*)(PRTL_OSVERSIONINFOEXW lpVersionInformation);
+
+        const auto ntModule{ ::GetModuleHandleW(L"ntdll.dll") };
+        if (!ntModule)
+            return;
+
+        RTL_OSVERSIONINFOEXW osVersion{ sizeof(RTL_OSVERSIONINFOEXW) };
+        const auto RtlGetVersion{
+            reinterpret_cast<fnRtlGetVersion>(::GetProcAddress(ntModule, "RtlGetVersion"))
+        };
+        if (!RtlGetVersion)
+            return;
         RtlGetVersion(&osVersion);
-        if (osVersion.dwMajorVersion > 0)
+        if (osVersion.dwMajorVersion > 0UL)
         {
-            sUserAgent.append("WindowsNT ");
-            sUserAgent.append(std::to_string(osVersion.dwMajorVersion));
-            sUserAgent.append(".");
-            sUserAgent.append(std::to_string(osVersion.dwMinorVersion));
+            // we don't have to explicitly reserve as long as the pointer isn't a nullptr
+            std::string str{ "" };
+            const auto needed
+            {
+                std::snprintf(str.data(), BUFSIZ, "WindowsNT %lu.%lu", osVersion.dwMajorVersion,
+                osVersion.dwMinorVersion)
+            };
+            if (needed > 0)
+                sUserAgent.append(str.c_str());
         }
     }
 
     sUserAgent.append(") Integration/");
+    {
+        std::string str{ "" };
+        const auto needed
+        {
+            std::snprintf(str.data(), BUFSIZ, "%d.%d.%d.%d", RA_INTEGRATION_VERSION_MAJOR,
```

Comment:
Same comment(s) about `sprintf`ing into a `std::string`. You need to make sure the string actually has the capacity specified as the buffer size, and need to update the length of the string afterwards.